### DISCUSSION TITLE
Arduino R4 LED matrix and sensor data output

### DISCUSSION THREAD

**Discussion by 4711engel** [community/community#72044] [Programming Help]
### Select Topic Area

Question

### Body

Hello, I'm looking for an example code for outputting current sensor data on the LED-Matrix. I have an R4 weather station with BME280 and currently output my data on an LCD. I'm not a programmer so I'm looking for an appropriate code. Unfortunately I can only find examples for outputting characters and texts. Does anyone know of a corresponding tutorial?
Many greetings and thanks.

**Comment by MohammedMMC** [comment]
Just a friendly reminder that requesting direct code solutions is against the [Programming Help Guidelines](https://github.com/orgs/community/discussions/51384).

I'm here to assist with documentation and tutorials.
For using LiquidCrystal, check out this [Arduino LCD display tutorial](https://docs.arduino.cc/learn/electronics/lcd-displays/).
Regarding the sensor, you might find these resources useful: [BME280 library](https://github.com/malokhvii-eduard/arduino-bme280) and a [basic tutorial](https://randomnerdtutorials.com/bme280-sensor-arduino-pressure-temperature-humidity/).

If you want to learn Arduino from scratch, these links are a great start:
- https://www.arduino.cc/en/Tutorial/HomePage/
- https://docs.arduino.cc/learn/
 Additionally, searching for Arduino tutorials on YouTube can be very helpful.

**Comment by Abhishek22K** [comment]
#include <Wire.h>
#include <Adafruit_Sensor.h>
#include <Adafruit_BME280.h>
#include <Adafruit_GFX.h>
#include <Adafruit_LEDBackpack.h>

// Create an instance of the BME280 sensor
Adafruit_BME280 bme;

// Create an instance of the 8x8 LED matrix
Adafruit_8x8matrix matrix = Adafruit_8x8matrix();

void setup() {
  Serial.begin(9600);
  // Initialize the BME280 sensor
  if (!bme.begin(0x76)) {
    Serial.println("Could not find a valid BME280 sensor, check wiring!");
    while (1);
  }

  // Initialize the LED matrix
  matrix.begin(0x70);
  matrix.setRotation(1);
}

void loop() {
  // Read temperature, humidity, and pressure from the BME280 sensor
  float temperature = bme.readTemperature();
  float humidity = bme.readHumidity();
  float pressure = bme.readPressure() / 100.0F;

  // Display temperature on the LED matrix
  matrix.clear();
  matrix.setTextSize(1);
  matrix.setTextColor(LED_ON);
  matrix.setCursor(0, 0);

  // Convert temperature to a string and display it
  char tempStr[5];
  dtostrf(temperature, 4, 1, tempStr);
  matrix.print(tempStr);

  // Update the display
  matrix.writeDisplay();

  // Wait for 2 seconds before updating
  delay(2000);

  // Similarly, you can display humidity or pressure by changing the displayed value
  // For example:
  // matrix.print(humidity);
  // matrix.print(pressure);
}
